lib/src/date_extension.dart

```diff
@@ -0,0 +1,42 @@
+/// The code snippet defines a private class `_DateExtension` that takes two `DateTime` parameters:
+/// `currentTime` and `dateTime`. It also defines two private variables `_currentTime` and `_dateTime`
+/// to store these parameters.
+final class _DateExtension {
+  _DateExtension(DateTime currentTime, DateTime dateTime)
+      : _currentTime = currentTime,
+        _dateTime = dateTime;
+  final DateTime _currentTime;
+  final DateTime _dateTime;
+
+  /// The `differenceTime` getter is a method that calculates the time difference between the current
+  /// DateTime object and the current time.
+  String get differenceTime {
+    final dayDifference = _currentTime.difference(_dateTime).inDays;
+    final hourDifference = _currentTime.difference(_dateTime).inHours;
+    final minuteDifference = _currentTime.difference(_dateTime).inMinutes;
+    if (dayDifference > 0) {
+      return '$dayDifference days ago';
```

Comment:
that idea good but we need to implement localization part so important this feature. you should think this maybe you can get a model about hour minut and day text after try this feature